sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -363,24 +374,31 @@ public Expression field(BlockBuilder list, int index, Type storageType) {
             Expressions.call(expression, "getValue", Expressions.constant(index)), Object.class);
       }
       FieldType fromType = inputSchema.getField(index).getType();
-      String getter = typeGetterMap.get(fromType.getTypeName());
+      String getter;
+      if (fromType.getTypeName().isLogicalType()) {
+        getter = logicalTypeGetterMap.get(fromType.getLogicalType().getIdentifier());
+      } else {
+        getter = typeGetterMap.get(fromType.getTypeName());
+      }
       if (getter == null) {
         throw new IllegalArgumentException("Unable to get " + fromType.getTypeName());
       }
-
       Expression field = Expressions.call(expression, getter, Expressions.constant(index));
-      if (fromType.getTypeName().isDateType()) {
+      if (fromType.getTypeName().isLogicalType()) {
         field = Expressions.call(field, "getMillis");
```

Comment:
This isn't going to work on just any `LogicalType` only `DateType`. Something is broken here.

Reply:
This should be inside the following if clauses I think (or alternatively wrapped with CalciteUtils.isDateTimeType). Do we have no tests that test CHAR type?